LSPS0/README.md

```diff
@@ -0,0 +1,606 @@
+> API endpoint to behave identically to older versions.
+
+Other LSPS specifications SHOULD specify some API call where LSPs
+can indicate which version(s) of that LSPS specification it
+supports, and SHOULD specify how clients can indicate which
+version it wnats to use.
+For example, an LSPS specification might indicate a "get
+information" API call that is stable across versions, which
+returns a `versions` array that indicates which versions of that
+LSPS specification the LSP supports.
+Other API calls then might include a `version` parameter
+that lets the client select which version of the LSPS
+specification it wants.
+
+#### Error Handling
+
+JSON-RPC 2.0 `error`s include a `message` field which is a
+human-readable error message.
+
+Clients MUST carefully filter error messages of possibly
+problematic characters, such as NUL characters, `<`, newlines,
+or ASCII control characters, which may cause problems when
+displaying in a dynamic context such as an HTML webpage, a
+TTY, or C string processing functions.
+
+Clients SHOULD NOT directly expose error `message`s to users
+unless users have enabled some kind of "advanced" mode or
+"developer" mode.
+Clients MAY write them in a log that is accessible to "advanced"
+or "developer" mode users.
+
+Clients SHOULD generate their own messages based on the
+error `code`, for display to the user, possibly integrating
+information in the optional `data` object if the error `code`
+defines it.
+If the client does not recognize the error `code`, or is
+expecting a field in the `data` object that is not present, it
+SHOULD indicate this as an "unrecognized" error to the user.
+
+> **Rationale** LSPs might write incorrect or misleading
+> human-readable error messages, and users might report such
+> error messages as bugs to the client developers, since the
+> user-visible source of the error message would be the client;
+> it is thus better if the client writes its own error
+> messages that it can change based on user feedback.
+> Newer versions of some LSPS spec may introduce new error codes,
+> or the specification may be incomplete and actual development
+> shows that some unspecified error could possibly occur, in
+> which case the human-readable `error` field could contain a
+> description of this error, which developers of clients can
+> then use to help guide the evolution of the specification, or
+> to comply with later versions of the LSPS spec.
+
+#### Disconnection Handling
+
+Networks are unreliable, and network participants are also unreliable.
+
+Clients MUST use a high-entropy `id` string for JSON-RPC 2.0 requests,
+such as a UUID, or a hex encoding of a random binary blob of at least
+80 bits, with randomness acquired from a cryptographically secure
+source.
+Clients MUST NOT use just a simple incrementing counter for the `id`.
+
+If a client receives a JSON-RPC 2.0 response with an `id` it does
+not remember sending a request for, it MUST ignore that response.
```

Comment:
> should be assigned an idempotency key

That is the responsibility of the other LSPS specs and not the transport layer IMO. For example, LSPS1 uses an orderId. This is can be persisted in case of a connection interruption.

> Or maybe it's better to add a note that the LSP MUST drop the request and any actions associated with it on failure to deliver the response within timeout.

You can't undo a channel open.